### DISCUSSION TITLE
Enable `OmegaConf.register_new_resolver` usage as decorator

### DISCUSSION THREAD

**Discussion by Jasha10** [omry/omegaconf#986] [Ideas]
Currently use of `register_new_resolver` requires (1) defining a function and (2) calling `register_new_resolver` on it. Enabling `register_new_resolver` to function as a decorator could simplify this workflow:

```python
...

@OmegaConf.register_new_resolver
def my_resolver(cfg):
    ...

# In this proposal, the above would be equivalent to the following:

def my_resolver(cfg):
    ...

OmegaConf.register_new_resolver("my_resolver", my_resolver)
```

**Comment by Northo** [comment]
I think this is a very good idea!
